src/Task.ts

```diff
@@ -428,7 +428,18 @@ export class Task {
             if (setCreatedDate) {
                 createdDate = window.moment();
             }
-            const nextStatus = StatusRegistry.getInstance().getNextStatusOrCreate(newStatus);
+            const statusRegistry = StatusRegistry.getInstance();
+            let nextStatus = statusRegistry.getNextStatusOrCreate(newStatus);
+            if (nextStatus.type !== StatusType.TODO) {
+                let searchStatus = nextStatus;
+                for (let i = 0; i < statusRegistry.registeredStatuses.length - 1; i++) {
+                    searchStatus = statusRegistry.getNextStatusOrCreate(searchStatus);
+                    if (searchStatus.type === StatusType.TODO) {
+                        nextStatus = searchStatus;
+                        break;
+                    }
+                }
+            }
```

Comment:
Thank you. The behaviour is a bit clearer now.

I will merge it in the next day or two.

I think a lot of readers would be confused as to why it loops over `statusRegistry.registeredStatuses.length` times... I certainly was.

So to help me maintain this code in future, and work out what extra tests to add, if you had time to add some comments, it would really help if you could please explain the algorithm.

Some things it would be helpful to explain:

- why does it look over `statusRegistry.registeredStatuses.length` times, instead of just going next, next, next...
- why is it using `getNextStatusOrCreate` rather than `getNextStatus()`?
- what happens if it cannot find any next status that is `TODO`.

Reply:
Great! Happy to help.

Here are some explanations for the implementation:

> why does it look over `statusRegistry.registeredStatuses.length` times, instead of just going next, next, next...

The goal is to avoid an infinite loop. By limiting the search to the number of configured statuses, we ensure it doesn't continue indefinitely.

> why is it using `getNextStatusOrCreate` rather than `getNextStatus()`?

There isn't a specific reason for this choice. I noticed it was used in the previous implementation, so I kept it. I can switch to ﻿`getNextStatus` if it makes more sense.

> what happens if it cannot find any next status that is `TODO`.

If it fails to find any ﻿`TODO` status, it will use the next symbol after ﻿`DONE`.

If you have more questions, please let me know

Reply:
Super - thank you very much indeed.